### PR TITLE
#168781713 - All users Mark Notifications

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?
Allows users to mark all their notification as read

#### Description of Task to be completed?
Enable an authenticated user to mark all their tasks as read.

#### How should this be manually tested?

1. Clone this repository
2. Run ` sequelize db:migrate `
3. Run ` sequelize db:seed:All `
4. Run ` npm run dev `
5. Log into the application at `api/v1/users/login`
6. Using an  authenticated user send a patch request to `api/v1/notifications/mark/read`

#### Any background context you want to provide?
N /A 
#### Screenshots (if appropriate)

Screen Shot 2019-11-04 at 17 06 41
Screen Shot 2019-11-04 at 17 06 31
Screen Shot 2019-11-04 at 17 09 42
Screen Shot 2019-11-04 at 17 09 46

#### Questions:

N / A

### CODE REVIEW

**Review by alainmateso — APPROVED** [review]
Gooj Job Cheza

**Review by nignanthomas — APPROVED** [review]
Great job @Cheza-Dzabala

File: src/controllers/notificationController.js [hunk 1]
```diff
@@ -8,12 +8,11 @@ const { NOTIF_NOT_FOUND, NOTIF_FOUND } = strings.notifications;
 export default class notificationController {
 
   static async allNotifications(req, res) {
-    const notifications = await allNotif(req.user.payload.id);
-
+    const query = { where: { userNotified: req.user.payload.id } };
+    const notifications = await allNotif(query);
```
Comment (nakiwuge) [inline src/controllers/notificationController.js]: Add a new line here

File: src/controllers/notificationController.js [hunk 2]
```diff
@@ -8,12 +8,11 @@ const { NOTIF_NOT_FOUND, NOTIF_FOUND } = strings.notifications;
 export default class notificationController {
 
   static async allNotifications(req, res) {
-    const notifications = await allNotif(req.user.payload.id);
-
+    const query = { where: { userNotified: req.user.payload.id } };
```
Comment (nakiwuge) [inline src/controllers/notificationController.js]: Kindly include the userObj

File: src/controllers/notificationController.js [hunk 3]
```diff
@@ -28,14 +27,44 @@ export default class notificationController {
...
+        userNotified,
+      }
+    };
 
-    if (!updatedNotif) {
-      return responseUtil(res, 404, NOTIF_NOT_FOUND);
+    try {
+      const updatedNotif = await updateNotif(query);
+      const status = (updatedNotif.isRead) ? '' : 'un';
+      const message = `This notification status has been marked as ${status}read`;
```
Comment (nakiwuge) [inline src/controllers/notificationController.js:45]: add anew line

File: src/controllers/notificationController.js [hunk 4]
```diff
@@ -28,14 +27,44 @@ export default class notificationController {
...
     }
+  }
 
-    const status = (updatedNotif.isRead) ? '' : 'un';
-    const message = `This notification status has been marked as '${status}read'`;
-    return responseUtil(res, 200, message);
+  static async markAllRead(req, res) {
+    const { id } = req.user.payload;
+    const query = { where: { userNotified: id, isRead: false } };
+    const notifications = await notifServices.allNotif(query);
```
Comment (nakiwuge) [inline src/controllers/notificationController.js:70]: add anew line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala good job. Kindly allow users to switch between `mark as read` and `mark as unread`.
Also, check out the comments I have left on the pr

File: src/controllers/notificationController.js [hunk 5]
```diff
@@ -28,14 +28,68 @@ export default class notificationController {
...
+      const status = (updatedNotif.isRead) ? '' : 'un';
+      const message = `This notification status has been marked as ${status}read`;
+      return responseUtil(res, 200, message);
+    } catch (error) {
+      return responseUtil(res, 404, error.message);
+    }
+  }
+
+  static async markAll(req, res) {
+
```
Comment (nakiwuge) [inline src/controllers/notificationController.js]: Remove this line

File: src/controllers/notificationController.js [hunk 6]
```diff
@@ -28,14 +28,68 @@ export default class notificationController {
...
+    let { action } = req.params;
+
```
Comment (nakiwuge) [inline src/controllers/notificationController.js]: Remove this line

File: src/controllers/notificationController.js [hunk 7]
```diff
@@ -28,14 +28,68 @@ export default class notificationController {
...
+      case 'unread':
+        isRead = false;
+        action = 'unread';
+        break;
+      default:
+        return responseUtil(res, 400, 'Unspecified action');
+    }
+
+    const { id } = req.user.payload;
+
```
Comment (nakiwuge) [inline src/controllers/notificationController.js]: Remove this line

File: src/controllers/notificationController.js [hunk 8]
```diff
@@ -28,14 +28,68 @@ export default class notificationController {
...
+    const query = { where: { userNotified: id, isRead: !isRead } };
+
```
Comment (nakiwuge) [inline src/controllers/notificationController.js]: Remove this line

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**